pkg/daemons/agent/agent_windows.go

```diff
@@ -2,5 +2,171 @@
 
 package agent
 
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/hcsshim"
+
+	"github.com/opencontainers/runc/libcontainer/system"
+	"github.com/rancher/k3s/pkg/cgroups"
+	"github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/util"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/kubernetes/pkg/kubeapiserver/authorizer/modes"
+)
+
 func createRootlessConfig(argsMap map[string]string, hasCfs, hasPIDs bool) {
 }
+
+func checkRuntimeEndpoint(cfg *config.Agent, argsMap map[string]string) {
+	if strings.HasPrefix(argsMap["container-runtime-endpoint"], windowsPrefix) {
+		argsMap["container-runtime-endpoint"] = cfg.RuntimeSocket
+	} else {
+		argsMap["container-runtime-endpoint"] = "npipe:////./pipe/containerd-containerd"
```

Comment:
Don't hardcode the path; if `cfg.RuntimeSocket` doesn't have the correct prefix then add the prefix but don't hardcode the whole thing here.

This exists because some of the kubelet args want `unix:///path` while others just want `/path`, and the user might have passed in either (if using their own container runtime endpoint instead of our packaged containerd). If the kubelet on windows is consistent about wanting `npipe:` on everything and `cfg.RuntimeSocket` is guaranteed to have this prefix, then you can probably just drop this check entirely.

```suggestion
		argsMap["container-runtime-endpoint"] = windowsPrefix + cfg.RuntimeSocket
```